collector/time.go

```diff
@@ -22,6 +22,9 @@ type TimeCollector struct {
 }
 
 func newTimeCollector() (Collector, error) {
+	if getWindowsVersion() < 10 {
+		log.Warn("Windows version older than Server 2016 detected. The time collector is likely to fail!")
```

Comment:
I'm considering if we shouldn't fail outright here? We know with high certainty that it will fail, and I'd personally rather have the system reject my settings in that case?

Reply:
Also, a vague recollection about Windows version numbers - does this actually work? IIRC you need to mark applications with a compatibility manifest to get the "real" version number, otherwise it'd lie and report an older version (8.1?).
I'm not 100% sure this applies here though, but I've been bitten by it in the past.

Reply:
>I'm considering if we shouldn't fail outright here?

Are we able to fail individual collectors? If so I imagine `log.Error()` would be suitable here?

Regarding the version numbers, I'm unfortunately unable to test this as I don't have a server older than 2016 available to me. Perhaps we could hold off on merging until someone's able to test this?

Reply:
Well, we can return an error, which would exit the exporter. I'd argue it is an invalid configuration to enable this collector on non-compatible machines?

Re version numbers, if we get back something higher than 6.1 on modern systems, we should be fine. Looks like the version masking might only relate to calling `GetVersion[Ex]` in the Windows APIs, while we read from the registry.

Reply:
Thanks Calle. I've updated the commit, could you let me know if it's alright?